service/history/shard/context_impl.go

```diff
@@ -1946,7 +1946,18 @@ func (s *ContextImpl) acquireShard() {
 	}
 
 	// keep retrying except ShardOwnershipLostError or lifecycle context ended
-	acquireShardRetryable := func(err error) bool {
+	acquireShardRetryable := func(err error) (isRetryable bool) {
+		defer func() {
+			msg := "This error is not retryable."
+			if isRetryable {
+				msg = "This error is retryable."
+			}
```

Comment:
Remove this, use const error message is easier to search for. The boolTag is all we need.

Reply:
Done